proxy/src/auth/backend/jwt.rs

```diff
@@ -0,0 +1,549 @@
+use std::{future::Future, sync::Arc, time::Duration};
+
+use anyhow::{bail, ensure, Context};
```

Comment:
After browsing the proxy codebase a bit more `thiserror` based errors seem preferred. Is the proxy being migrated to `anyhow` or is this temporary?

Reply:
It's intentionally hybrid - they compliment each other. I was being lazy in this code using anyhow, I wanted to make it work first and think about error handling later. Eventually I'll move over to typed errors, as they integrate better with our error classification system.

Long term I've wanted to move to error-stack but haven't had the time to properly invest in a proof of concept